Hudl.Mjolnir/Util/ConcurrentDictionaryExtensions.cs

```diff
@@ -12,5 +12,11 @@ internal static class ConcurrentDictionaryExtensions
             var lazy = dictionary.GetOrAdd(key, new Lazy<V>(() => valueFactory(key), LazyThreadSafetyMode.PublicationOnly));
```

Comment:
Yeah, I might even get rid of this one altogether and have the caller make a decision about the concurrency they'd like.